src/bin/ambit/cmd.rs

```diff
@@ -0,0 +1,191 @@
+        return Err(AmbitError::Other(
+            "Dotfile repository already exists.\nUse '-f' flag to overwrite.".to_owned(),
+        ));
+    } else if AMBIT_PATHS.repo.exists() {
+        // Repository directory exists but force is enabled
+        AMBIT_PATHS.repo.remove()?;
+    }
+    Ok(())
+}
+
+// Fetch entries from config file and return as vector
+fn get_config_entries() -> AmbitResult<Vec<config::Entry>> {
+    let content = AMBIT_PATHS.config.as_string()?;
+    config::get_entries(content.chars().peekable())
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(AmbitError::Parse)
+}
+
+// Initialize an empty dotfile repository
+pub fn init(force: bool) -> AmbitResult<()> {
+    ensure_paths_exist(force)?;
+    AMBIT_PATHS.repo.create()?;
+    // Initialize an empty git repository
+    git(vec!["init"])?;
+    Ok(())
+}
+
+// Clone an existing dotfile repository with given origin
+pub fn clone(force: bool, arguments: Vec<&str>) -> AmbitResult<()> {
+    ensure_paths_exist(force)?;
+    // Clone will handle creating the repository directory
+    let repo_path = AMBIT_PATHS.repo.to_str()?;
+    let status = Command::new("git")
+        .arg("clone")
+        .args(arguments)
+        // Pass in ambit repo path as last argument to ensure that it is always cloned to the known path
+        .arg(repo_path)
+        .status()?;
+    match status.success() {
+        true => {
+            println!("Successfully cloned repository to {}", repo_path);
+            Ok(())
+        }
+        false => Err(AmbitError::Other("Failed to clone repository".to_owned())),
+    }
+}
+
+// Check ambit configuration for errors
+pub fn check() -> AmbitResult<()> {
+    get_config_entries()?;
+    Ok(())
+}
+
+// Sync files in dotfile repository to system through symbolic links
+pub fn sync(dry_run: bool, quiet: bool) -> AmbitResult<()> {
+    // Only symlink if repo and git directories exist
+    if !(AMBIT_PATHS.repo.exists() && AMBIT_PATHS.git.exists()) {
+        return Err(AmbitError::Other(
+            "Dotfile repository does not exist. Run `init` or `clone` before syncing.".to_owned(),
+        ));
+    }
+    let mut successful_symlinks = 0; // Number of symlinks that actually occurred
+    let mut total_symlinks = 0;
+    // link is a closure as it must have access to outside variables
+    let mut link = |repo_filename: &str, host_filename: &str| -> AmbitResult<()> {
+        let host_file = AmbitPath::new(
+            AMBIT_PATHS.home.path.join(host_filename),
+            AmbitPathKind::File,
+        );
+        let host_file_str = host_file.to_str()?;
+        let repo_file = AmbitPath::new(
+            AMBIT_PATHS.repo.path.join(repo_filename),
+            AmbitPathKind::File,
+        );
+        let repo_file_str = repo_file.to_str()?;
+
+        // already_symlinked holds whether host_file already links to repo_file
+        let already_symlinked = if let Ok(link_path) = fs::read_link(&host_file.path) {
```

Comment:
This can be written more concisely by the use of `.map`, e.g.
```rust
fs::read_link(&host_file.path).map(|link_path| link_path == repo_file.path).unwrap_or(false)
```